### PR TITLE
Remove U2F support

### PR DESCRIPTION
**Pull request by codingllama**
Follows up on #10466 by removing remaining U2F references, including proto/gRPC surface and the lib/auth/u2f package itself.

#10375

### CODE REVIEW

**Comment by codingllama** [comment]
Branched from #10375, once that's merged the target automatically switches to master.

Looks gigantic in lines added because of proto gen. Deleted lines we do have a lot. :)

FYI @zmb3, in case you want to review both parts.

File: api/client/proto/authservice.proto [hunk 1]
```diff
@@ -747,9 +746,7 @@ enum DeviceType {
 // MFAAuthenticateChallenge is a challenge for all MFA devices registered for a
 // user.
 message MFAAuthenticateChallenge {
-    // U2F contains one U2FChallenge per U2F device registered for a
-    // user. Each challenge is unique.
-    repeated U2FChallenge U2F = 1;
+    reserved 1; // repeated U2FChallenge U2F
```
Comment (kimlisa) [inline api/client/proto/authservice.proto:749]: just curious why this one has to be reserved?

File: lib/auth/password.go [hunk 2]
```diff
@@ -132,10 +117,8 @@ func (s *Server) ChangePassword(req services.ChangePasswordReq) error {
 			Password: req.OldPassword,
 		}
 	}
-	if req.U2FSignResponse != nil {
-		authReq.U2F = &U2FSignResponseCreds{
-			SignResponse: *req.U2FSignResponse,
-		}
+	if req.WebauthnResponse != nil {
+		authReq.Webauthn = req.WebauthnResponse
```
Comment (kimlisa) [inline lib/auth/password.go:121]: i was questioning how this was working before :laughing:, but this is already set in line 113, remove one of them?

**Review by kimlisa — APPROVED** [review]

File: api/client/proto/authservice.proto (hunk 1 above)
Comment (codingllama) [reply api/client/proto/authservice.proto:749]: `reserved` is so the field tag can't be reused. I'm mostly playing it safe, but tag reuses can cause bugs that are difficult to reason about. If server and client drift in version they may think the same tag represents different fields/types, likely cause somewhat cryptic gRPC errors (or, in worst case scenario, bogus data).

Entire messages are safe to be deleted. For fields, I've reserved all I could, except for oneof fields, which can't be reserved.

File: lib/auth/password.go (hunk 2 above)
Comment (codingllama) [reply lib/auth/password.go:121]: That explains my confusion too. Done.

**Comment by codingllama** [comment]
We're totally up-to-date with master now and all prereqs have landed. I've also updated e/ to https://github.com/gravitational/teleport.e/commit/ad8fd4aa9128b3dd5e80e84804be9850004f58b1 as part of the PR, so it should be pristine now.

PTAL?

**Comment by codingllama** [comment]
For reference, I did a small migration test and things seem to be working fine. The scenario is something like:

1. Create a fresh Teleport v9 cluster, set "second_factor:optional" and "webauthn.disabled:true" (aka force U2F)
1. Create a user, sign up with OTP, add an U2F device, login in both tsh and Web using U2F
1. Stop Teleport, migrate to PR version and restart
1. Logout and login in both tsh and Web using the registered U2F device, register a new WebAuthn device in both interfaces, login with new device

**Review by smallinsky — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**